### PR TITLE
Fix login logic at app initialization time.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6582/

**Comment by bac** [comment]
Looks reasonable.

File: jujugui/static/gui/src/app/components/logout/logout.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ YUI.add('logout-component', function() {
       getUser: React.PropTypes.func.isRequired,
       gisf: React.PropTypes.bool.isRequired,
       gisfLogout: React.PropTypes.string.isRequired,
-      locationAssign: React.PropTypes.bool.isRequired,
+      locationAssign: React.PropTypes.func.isRequired,
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/logout/logout.js]: Nice cleanup, thanks!

File: jujugui/static/gui/src/app/components/logout/logout.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ YUI.add('logout-component', function() {
       props.clearCookie();
       props.logout();
       if (props.gisf) {
-        props.locationAssign(window.location.origin + this.props.gisfLogout); 
+        props.locationAssign(window.location.origin + this.props.gisfLogout);
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/logout/logout.js]: Ditto on the cleanup.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**